File: tests/fixtures/codegen/expected/typescript/runtime/queries/events/GetBillingEvent.ts
import type { GetBillingEventParams, GetBillingEventRow } from "../../../types/queries/events/GetBillingEvent.js";
export type { GetBillingEventParams, GetBillingEventRow } from "../../../types/queries/events/GetBillingEvent.js";
import type * as pgsid from "./pgsid/queryable.js";
export const getBillingEventSql = "SELECT id, source_id, state FROM billing.events WHERE id = $1;";
export async function getBillingEvent(db: pgsid.Queryable, params: GetBillingEventParams): Promise<GetBillingEventRow | undefined> {
    const result = await db.query(getBillingEventSql, [params["id"]]);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    return row as GetBillingEventRow;
}

File: tests/fixtures/codegen/expected/go/queries/events/getbillingevent.go
package events

import (
	"context"
	billing "example.com/pgsid-fixture/generated/go/schema/billing"
	public "example.com/pgsid-fixture/generated/go/schema/public"
)

const GetBillingEventSQL = "SELECT id, source_id, state FROM billing.events WHERE id = $1;"

type GetBillingEventParams struct {
	Id *int64 `db:"id"`
}
type GetBillingEventRow struct {
	Id       billing.EventId    `db:"id"`
	SourceId public.EventId     `db:"source_id"`
	State    billing.EventState `db:"state"`
}

func (q *Queries) GetBillingEvent(ctx context.Context, params GetBillingEventParams) (GetBillingEventRow, error) {
	var row GetBillingEventRow
	err := q.db.QueryRow(ctx, GetBillingEventSQL, params.Id).Scan(&row.Id, &row.SourceId, &row.State)
	return row, err
}
